scripts/apps/authoring/macros/macros.ts

```diff
@@ -81,7 +81,11 @@ function MacrosService(api, notify) {
             commit: !!commit,
         }).then((res) => res, (err) => {
             if (angular.isDefined(err.data._message)) {
-                notify.error(gettext('Error: {{message}}', {message: err.data._message}));
+                const error_messages = JSON.parse(err.data._message.replace(/'/g, '"'));
```

Comment:
Why is it needed to replace quotes? Does the error come as a invalid JSON string? Could it come as an object/array and not a string?

Reply:
yes, there was an error while doing JSON parse. But now I have handled it in the back end itself.